### PR TITLE
Add Focus Flow

### PR DESCRIPTION
**Pull request by pranavbehal**
# BrowserBuddy Extension Submission

## Project Details

1. **What is the name of your project?**
Focus Flow

2. **How many hours did you spend developing this?**  
   _(Give your best estimate)_
15-20 hours.

3. **What problem does your extension solve, or what makes it unique?**  
   _(Provide a brief description of the functionality or innovation)_
This extension is a website blocker to boost your productivity, but it uses AI to block websites, rather than just having a static list of sites to block/not block. In the past, I tried using basic site blockers to reduce distractions, but they were always easy to take advantage of. I made this so that AI can dynamically let you use websites that align with a task you set to complete during a session (with a Pomodoro timer), which helps boost your productivity and reduce distractions. If the AI does make a mistake, you always have the option to make an appeal to access a website. From my testing, I've seen that my AI prompt usually makes the right decision on the first try (without an appeal).

4. **How many people have tested your extension?**  
   _(We recommend at least 3)_
7 people.

5. **Does your extension work in both Chrome and Firefox?**  
   _(Your extension must be compatible with both browsers)_
My extension works on Chrome & Edge, but I haven't tested Firefox as I don't have it. However, my extension's version is Manifest 3.

---

## Additional Comments
**In order to use the extension, you must add your own OpenAI API key in the api-key.js file.**

Also, here is a demo video I made (I remember reading somewhere that we needed one, but I wasn't sure and made one anyways):
https://drive.google.com/file/d/1ZEErvjxo0wj5tDVqKNM54SptbaOgdQGc/view?usp=sharing

### CODE REVIEW

File: focus-flow/api-key.js
Comment (JosiasAurel) [inline focus-flow/api-key.js:1]: end users won't be able to modify the source code of the extension once published. you should add an interface in the extension where users can set their OpenAI API Key

**Review by JosiasAurel — CHANGES_REQUESTED** [review]
Please add an option to set the OpenAI API key in the extension along with instructions on how to get one. Otherwise I think this is a cool extension.

Also please add your extensions in the `submissions` folder.

File: focus-flow/api-key.js
Comment (pranavbehal) [reply focus-flow/api-key.js:1]: I made another commit so that you can add your own API key in the "Settings' tab of the extension, please take a look.

**Review by JosiasAurel — APPROVED** [review]
Awesome! If you haven't yet, please fill [this form](https://airtable.com/appgtKZwZBoOeFU7O/pagyeS51zr2kC1rp0/form) to get your grant.

**Pull request merged**